contribs/gmf/src/datasource/Manager.js

```diff
@@ -199,7 +207,50 @@ const exports = class {
    *     object to keep a reference of in this service.
    */
   setDimensions(dimensions) {
+    if (this.dimensionsWatcherUnregister) {
+      this.dimensionsWatcherUnregister();
+    }
+
     this.dimensions_ = dimensions;
+
+    this.dimensionsWatcherUnregister = this.rootScope_.$watch(
+      () => this.dimensions_,
+      this.handleDimensionsChange_.bind(this),
+      true
+    );
+    this.handleDimensionsChange_();
```

Comment:
handleDimensionsChange_ is only called here. So why do you need this watch, this register, etc... ?

Reply:
dimensions object is only set here, but we need to watch changes in dimensions objet properties ex:
{ "FLOOR": "+1" }

Reply:
Oh, I undertsand. Thanks.
But then, are you sure it works with a simple `$watch` ? ? Don't you need a `$watchCollection`  Because I think that the reference will never change. Otherwise, it's more strange to have a private object that change like that.

Reply:
This works using the third parameter : true.
$watch(watchExpression, listener, [objectEquality]);
Property was already there, I just added a watcher to update layer's FILTER parameter.
this.dimensions_ is the same object as main controller dimensions property.